lib/src/element_type.dart

```diff
@@ -23,7 +23,11 @@ class ElementType {
 
   bool get isParameterizedType {
     if (_type is FunctionType) {
-      return (_type as FunctionType).typeFormals.isNotEmpty;
+      final FunctionType functionType = _type;
+      final Iterable<DartType> typeArguments = []
+          ..addAll(functionType.typeArguments)
+          ..addAll(functionType.typeFormals.map((f) => f.type));
+      return typeArguments.isNotEmpty;
     } else if (_type is ParameterizedType) {
```

Comment:
@bwilkerson, do these changes look ok? As mentioned, this fixes the issue with generic info not showing up for typdef return types.